lib/orderbook/OrderBook.ts

```diff
@@ -345,6 +352,53 @@ class OrderBook extends EventEmitter {
     return outgoingOrder;
   }
 
+  /**
+   * Handles a request from a peer to create a swap deal. Checks if the order for the requested swap
+   * is available and if a route exists to determine if the request should be accepted or rejected.
+   * Responds to the peer with a swap response packet containing either an accepted quantity or rejection reason.
+   */
+  private handleSwapRequest = async (requestPacket: SwapRequestPacket, peer: Peer)  => {
+    assert(requestPacket.body, 'SwapRequestPacket does not contain a body');
+    assert(this.swaps, 'swaps module is disabled');
+    let rejectionReason: SwapRejectionReason | undefined;
+    let acceptedQuantity = 0;
+    const requestBody = requestPacket.body!;
+
+    if (!this.matchingEngines.has(requestBody.pairId)) {
+      rejectionReason = SwapRejectionReason.PAIR_NOT_SUPPORTED;
+    } else {
+      const order = this.getOwnOrder(requestBody.orderId, requestBody.pairId);
+      if (!order) {
+        rejectionReason = SwapRejectionReason.ORDER_NOT_FOUND;
+      } else {
+        const availableQuantity = order.hold ? order.quantity - order.hold : order.quantity;
+        if (availableQuantity === 0) {
+          rejectionReason = SwapRejectionReason.ORDER_UNAVAILABLE;
+        } else {
+          // accept the smaller of the proposed quantity and the available quantity
+          acceptedQuantity = Math.min(requestBody.proposedQuantity, availableQuantity);
+          // check that a route exists to pay our part of the swap
+          const hasRoute = await this.swaps!.acceptDeal(acceptedQuantity, order.price, requestBody, peer);
```

Comment:
I was thinking this wasn't necessary because it's single threaded, but since `acceptDeal` is async it's possible that we handle another swap request while waiting on the `queryRoutes` response. Will fix.